src/mono/netcore/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Threading
+{
+    public sealed class RegisteredWaitHandle : MarshalByRefObject
+    {
+        internal RegisteredWaitHandle(WaitHandle waitHandle, _ThreadPoolWaitOrTimerCallback callbackHelper,
+            int millisecondsTimeout, bool repeating)
+        {
+        }
+
+        public bool Unregister(WaitHandle? waitObject)
+        {
+            throw new PlatformNotSupportedException();
+        }
+    }
+
+    public static partial class ThreadPool
+    {
+        internal const bool EnableWorkerTracking = false;
+
+        private static bool _callbackQueued;
+
+        internal static void InitializeForThreadPoolThread() { }
+
+        public static bool SetMaxThreads(int workerThreads, int completionPortThreads)
+        {
+            if (workerThreads == 1 && completionPortThreads == 1)
+                return true;
+            return false;
+        }
+
+        public static void GetMaxThreads(out int workerThreads, out int completionPortThreads)
+        {
+            workerThreads = 1;
+            completionPortThreads = 1;
+        }
+
+        public static bool SetMinThreads(int workerThreads, int completionPortThreads)
+        {
+            if (workerThreads == 1 && completionPortThreads == 1)
+                return true;
+            return false;
+        }
+
+        public static void GetMinThreads(out int workerThreads, out int completionPortThreads)
+        {
+            workerThreads = 1;
+            completionPortThreads = 1;
+        }
+
+        public static void GetAvailableThreads(out int workerThreads, out int completionPortThreads)
+        {
+            workerThreads = 1;
+            completionPortThreads = 1;
+        }
+
+        public static int ThreadCount => 1;
+
+        public static long CompletedWorkItemCount => 0;
+
+        internal static void RequestWorkerThread()
+        {
+            if (_callbackQueued)
+                return;
+            _callbackQueued = true;
+            QueueCallback();
+        }
+
+        internal static bool KeepDispatching(int startTickCount)
+        {
+            return true;
+        }
+
+        internal static void NotifyWorkItemProgress()
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static bool NotifyWorkItemComplete()
+        {
+            return false;
```

Comment:
Do you intend for Dispatch() to process only one work item at a time? Curious why this would not return true instead, to process all queued work items.